src/sfdxFlags.ts

```diff
@@ -700,6 +694,17 @@ function isBuiltin(flag: object): flag is flags.Builtin {
   return hasString(flag, 'type') && flag.type === 'builtin';
 }
 
+type Output = {
+  json: flags.Discriminated<flags.Boolean<boolean>>;
```

Comment:
some of these are already defined somewhere else.  Example...you could refer to the place that you're going to get them.

```ts
  apiversion?: ReturnType<typeof optionalBuiltinFlags['apiversion']>;
```

if you're feeling really ambitious, you could probably find some way to not reference each of them individually, but you might want to anyway to define which are required vs. optional